File: Cap 05/5.5 - Laco Condicional Seletivo/Efetivo/laco.rb
=begin
 * Livro .....: Algoritmos
 *              Logica para Desenvolvimento de Programacao de
 *              Computadores
 * Autores ...: Jose Augusto N. G. Manzano
 *              Jayr Figueiredo de Oliveira
 * Editora ...: Erica
 * 
 * Capitulo ..: 5 - Topico 5.6
 * 
 * Programa ..: LACO SELETIVO ITERATIVO
=end

puts 'LACO PRE-TESTE COM FLUXO VERDADEIRO'
puts 'Multiplicacao de numero por 3 por 5 vezes'
puts
i = 1
loop do # equivalente a "laco"
  puts "Calculo: #{i}"
  print 'Entre valor numerico inteiro: '
  n = gets.chomp.to_i
  r = n * 3
  puts "O numero informado X 3 = #{r}"
  puts
  break if (i > 4) # equivalente a "saia_caso (I > 4)"
  i = i + 1
end # equivalente a "fim_laco"

File: Cap 05/5.5 - Laco Condicional Seletivo/Efetivo/laco.vb
' Livro .....: Algorimos
'              Logica para Desenvolvimento de Programacao de 
'              Computadores
' Autores ...: Jose Augusto N. G. Manzano
'              Jayr Figueiredo de Oliveira
' Editora ...: Erica
' 
' Capitulo ..: 5 - Topico 5.5
' 
' Programa ..: LACO SELETIVO ITERATIVO

Module LoopSelectiveIterative

    Sub Main()

        Dim N As Integer
        Dim R As Integer
        Dim I As Integer
        Console.WriteLine("LACO SELETIVO ITERATIVO")
        Console.WriteLine("Multiplicacao de numero por 3 por 5 vezes")
        Console.WriteLine()
        I = 1
        Do
            Console.WriteLine("Calculo: " & I)
            Console.Write("Entre valor numerico inteiro: ")
            N = Console.ReadLine()
            R = N * 3
            Console.WriteLine("O numero informado X 3 = " & R)
            Console.WriteLine()
            If (I > 4) Then Exit Do
            I = I + 1
        Loop

    End Sub

End Module
